### ISSUE TITLE
Support custom request methods (such as vault's LIST request)

### ISSUE THREAD

**Issue by paul-at-cybr** [usebruno/bruno#1473]
LIST is used as the primary request method on the Vault API's [List Roles](https://developer.hashicorp.com/vault/api-docs/auth/kubernetes#list-roles) endpoint.

A GET alternative is available since they know they're being weird, but it might be nice to support custom request methods on a general basis.

**Comment by DutchForeigner** [comment]
In my work place, we are required to move away from Postman (to Bruno). The challenge for us is indeed that the LIST method is used with HashiCorp Vault. As a result we could not import our Vault collections from Postman (without renaming each LIST entry to GET) and existing 'Requests' with List are now not working.

So I would 2nd this Feature request to add custom methods similar to what is in Postman
![image](https://github.com/usebruno/bruno/assets/76488714/289393f6-b141-4684-9b1d-a7373094abfe)

**Comment by Its-treason** [comment]
As a workaround: You can set custom request methods inside the pre-request-script:

```js
req.setMethod("SEARCH")
```

**Comment by rubancar** [comment]
any update on this? I'm in the same situation needing the method LIST.

**Comment by Its-treason** [comment]
> any update on this? I'm in the same situation needing the method LIST.

Nope, you still have to use the workaround, I mentioned in the previous comment

**Comment by DutchForeigner** [comment]
FYI: For people who need the LIST method for use with HashiCorp Vault, you can replace LIST with **GET** + add **?list=true** to the URL (e.g. add a query to it)

**Comment by Rockybilly** [comment]
I also need this for the PURGE method. Although it's not standard, I believe it's crazy to not have custom http method support for an app such as Bruno

**Comment by paul-at-cybr** [comment]
🎉 🎉

**Comment by helloanoop** [comment]
PR has been merged! This will be available in our upcoming `2.11.0` release scheduled to go out on 9th Sep 2025

https://github.com/user-attachments/assets/23b2d1cb-e9fc-4d26-9f1e-5db33a8aa771
